lib/Models/NominatimSearchProvider.ts

```diff
@@ -0,0 +1,166 @@
+
+    const promiseBounded: Promise<any> = loadJson(
+      new Resource({
+        url: this.url + "search",
+        queryParameters: {
+          q: searchText,
+          limit: this.limitOthers,
+          countryCodes: this.countryCodes,
+          format: "json",
+          bounded: 1,
+          viewbox: bbox.join(",")
+        }
+      })
+    );
+
+    const promiseOthers: Promise<any> = loadJson(
+      new Resource({
+        url: this.url + "search",
+        queryParameters: {
+          q: searchText,
+          limit: this.limitOthers,
+          countryCodes: this.countryCodes.join(","),
+          format: "json"
+        }
+      })
+    );
+
+    return when.all([promiseBounded, promiseOthers]).then((result: any) => {
+      if (searchResults.isCanceled) {
+        // A new search has superseded this one, so ignore the result.
+        return;
+      }
+
+      if (result.length === 0) {
+        searchResults.message = i18next.t("viewModels.searchNoLocations");
+        return;
+      }
+      const locations: SearchResult[] = [];
+      // Locations in the bounded query go on top, locations elsewhere go undernearth
+      const findDbl = function(locations: SearchResult[], place_id: number) {
+        return locations.filter(function(location) {
+          return location.id === place_id;
+        })[0];
+      };
+
+      for (let i = 0; i < result.length; ++i) {
+        for (let j = 0; j < result[i].length; ++j) {
+          const resource = result[i][j];
+          var name = resource.display_name;
+          if (!defined(name)) {
+            continue;
+          }
+
+          if (!findDbl(locations, resource.place_id)) {
+            locations.push(
+              new SearchResult({
+                id: resource.place_id,
+                name: name,
+                isImportant: true,
+                clickAction: createZoomToFunction(this, resource),
+                location: {
+                  latitude: resource.lat,
+                  longitude: resource.lon
+                }
+              })
+            );
+          }
+        }
+      }
+      runInAction(() => {
+        searchResults.results.push(...locations);
+      });
+
+      if (searchResults.results.length === 0) {
+        searchResults.message = i18next.t("viewModels.searchNoLocations");
+      }
+    });
+  }
+}
+
+function createZoomToFunction(model: NominatimSearchProvider, resource: any) {
+  const [south, north, west, east] = resource.boundingbox;
+  const rectangle = Rectangle.fromDegrees(west, south, east, north);
+
+  return function() {
+    const terria = model.terria;
+    terria.currentViewer.zoomTo(rectangle, model.flightDurationSeconds);
+  };
+}
+
+export class ExtendedNominatimSearchProvider extends NominatimSearchProvider {
+  doSearch(
+    searchText: string,
+    searchResults: SearchProviderResults
+  ): Promise<void> {
+    return super.doSearch(searchText, searchResults);
+  }
+}
```

Comment:
@zoran995 I think it would be better to test this provider through the public `search` function, rather than `doSearch`. Is there a reason this can't be done?

If it can't be done that way then this test helper should be moved into the test file.

Reply:
the only way that I saw that could be possible with `search` was to add some timeout so I moved to doSearch.

Reply:
moved this to test file